Library/Homebrew/test/test_utils.rb

```diff
@@ -210,4 +210,17 @@ def test_number_readable
     assert_equal "1,000", number_readable(1_000)
     assert_equal "1,000,000", number_readable(1_000_000)
   end
+
+  def test_truncate_text_to_approximate_size
+    glue = "\n[...snip...]\n" # hard-coded copy from truncate_text_to_approximate_size
+    n = 20
+    long_s = "x" * 40
+    s = truncate_text_to_approximate_size(long_s, n)
+    assert_equal s.length, n
+    assert_match(/x+#{Regexp.escape(glue)}x+/, s)
```

Comment:
Minor nit, but if you wanted to make sure there's nothing else other than `x`es before and after the glue, you'd need to anchor the regular expression at the start and end of the string using `^` and `$`. If you just want to check that the glue appears somewhere in the middle (i.e. there is at least one `x` before and after it), then the following is basically equivalent to the current check, but simpler:

```ruby
assert_match "x#{glue}x", s
```